tests/test_helper.py

```diff
@@ -31,55 +43,50 @@ def test_gen_random_alphabet_string(self):
-            ('def', 'a'), ('bc', 'def'), ('def', 'bc'), ('a', 'bc', 'def'),
-            ('a', 'def', 'bc'), ('bc', 'a', 'def'), ('bc', 'def', 'a'),
-            ('def', 'a', 'bc'), ('def', 'bc', 'a')
-        ])
+        errors = []
+        data = [
+            '',
+            'a',
+            'ab',
+            'abc',
+            [],
+            ['a'],
+            ['a', 'bc'],
+            ['a', 'bc', 'def'],
+        ]
+        expect = [
+            [],
+            [('a', )],
+            [('a', ), ('b', ), ('a', 'b'), ('b', 'a')],
+            [('a', ), ('b', ), ('c', ), ('a', 'b'), ('b', 'a'), ('a', 'c'), ('c', 'a'), ('b', 'c'), ('c', 'b'), ('a', 'b', 'c'), ('a', 'c', 'b'), ('b', 'a', 'c'), ('b', 'c', 'a'), ('c', 'a', 'b'), ('c', 'b', 'a')],  # noqa: E501
+            [],
+            [('a', )],
+            [('a', ), ('bc', ), ('a', 'bc'), ('bc', 'a')],
+            [('a', ), ('bc', ), ('def', ), ('a', 'bc'), ('bc', 'a'), ('a', 'def'), ('def', 'a'), ('bc', 'def'), ('def', 'bc'), ('a', 'bc', 'def'), ('a', 'def', 'bc'), ('bc', 'a', 'def'), ('bc', 'def', 'a'), ('def', 'a', 'bc'), ('def', 'bc', 'a')],  # noqa: E501
+        ]
+
+        actual = []
+        for datum in data:
+            actual.append(gen_all_possible_pair(datum))
+        for i in range(0, len(data)):
+            helper.append_to_error_if_not_expect_with_msg(
+                errors,
+                expect[i] == actual[i],
+                "Expect:\n{0}\nActual:\n{1}".format(expect[i], actual[i]))
+
+        assert errors == [], Exception(helper.get_error_string(errors))
 
     def test_get_all_type_values_of_json(self):
+        # only test values is valid or else.
         get_all_type_values_of_json = helper.get_all_type_values_of_json
         errors = []
-        # int, float, True, False, None, dict, list, (long), (str), (unicode)
-        expect_types = (int, float, True, False, None, dict, list)
-        # Compatibility
-        try:  # pragma: no cover
-            expect_types = (long, ) + expect_types  # noqa: F821
-            expect_types = (unicode, ) + expect_types  # noqa: F821
-        except Exception:  # pragma: no cover
-            expect_types = (str, ) + expect_types
-
-        actual_type_values = get_all_type_values_of_json()
-        for expect in expect_types:
-            has_type = False
-            for v in actual_type_values:
-                if (v == expect and
-                   (expect is None or expect is True or expect is False)):
-                    has_type = True
-                elif type(v) == expect:
-                    has_type = True
-            helper.append_to_error_if_not_expect_with_msg(
-                errors,
-                has_type,
-                "the type '{0}' hasn't existed.".format(expect)
-            )
+
+        actual = get_all_type_values_of_json()
+        for actl in actual:
+            try:
+                json.loads(json.dumps(actl))
+            except Exception:
```

Comment:
I _wouldn't_ catch the generic exception, only a specific one.

Reply:
I want to catch **all** error.
And since the functions might raise several errors, and it may be updated in future, thus I use generic exception, otherwise the flake8 will complain.